vendor/bat-native-ads/src/bat/ads/internal/filters/ad_history_confirmation_filter.cc

```diff
@@ -0,0 +1,91 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+#include <algorithm>
+#include <memory>
+#include <string>
+
+#include "bat/ads/internal/filters/ad_history_confirmation_filter.h"
+#include "bat/ads/confirmation_type.h"
+#include "bat/ads/ad_history_detail.h"
+
+#include "bat/ads/ads_history.h"
+
+namespace ads {
+
+bool IsConfirmationTypeOfInterest(const ConfirmationType& confirmationType) {
+  bool isOfInterest = false;
+
+  if ((confirmationType == ads::ConfirmationType::Value::CLICK)
+      || (confirmationType == ads::ConfirmationType::Value::VIEW)
+      || (confirmationType == ads::ConfirmationType::Value::DISMISS)) {
+    isOfInterest = true;
+  }
+
+  return isOfInterest;
+}
+
+bool DoesConfirmationTypeATrumpB(const ConfirmationType& confirmationTypeA,
+    const ConfirmationType& confirmationTypeB) {
+  bool doesTypeATrumpTypeB = false;
+  if (confirmationTypeA == ads::ConfirmationType::Value::CLICK) {
+    if ((confirmationTypeB == ads::ConfirmationType::Value::CLICK)
+        || (confirmationTypeB == ads::ConfirmationType::Value::VIEW)
+        || (confirmationTypeB == ads::ConfirmationType::Value::DISMISS)) {
+      doesTypeATrumpTypeB = true;
+    }
+  } else if (confirmationTypeA == ads::ConfirmationType::Value::VIEW) {
+    if ((confirmationTypeB == ads::ConfirmationType::Value::VIEW)
+        || (confirmationTypeB == ads::ConfirmationType::Value::DISMISS)) {
+      doesTypeATrumpTypeB = true;
+    }
+  } else if (confirmationTypeA == ads::ConfirmationType::Value::DISMISS) {
+    if (confirmationTypeB == ads::ConfirmationType::Value::DISMISS) {
+      doesTypeATrumpTypeB = true;
+    }
+  }
+
+  return doesTypeATrumpTypeB;
+}
+
+AdHistoryConfirmationFilter::~AdHistoryConfirmationFilter() = default;
+
+std::deque<AdHistoryDetail> AdHistoryConfirmationFilter::ApplyFilter(
+    const std::deque<AdHistoryDetail>& adsHistory) const {
+
+  std::map<std::string, AdHistoryDetail> filteredAdHistory;

```

Comment:
`filteredAdHistory ` should use `camel_case`

Reply:
Done